server/api/auth.ts

```diff
@@ -0,0 +1,46 @@
+import express from "express";
+
+import { Sessions, Users } from "../models";
+import { generateToken } from "../security";
+
+const authApi = express.Router();
+
+authApi.post("/sign-in-token", async (req, res) => {
```

Comment:
I did consider this at one point, deemed it strange to send a cookie with every single request and implemented it as an extra endpoint. Thinking about it again, particularly if it's limited to `index.html` as you suggest, it is much more elegant 👌

Reply:
... of course, this probably does mean it won't work in the dev environment since webpack is serving the files there, but, naturally, that shouldn't influence the implementation.